src/tenant/forms.py

```diff
@@ -55,3 +59,91 @@ class TenantForm(TenantBaseForm):
 
     class Meta(TenantBaseForm.Meta):
         fields = TenantBaseForm.Meta.fields + ["first_name", "last_name", "email"]
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the TenantForm.
+
+        If `verified_data` is provided in kwargs, pre-fill the form fields
+        from it and disable the email field so that it cannot be modified.
+
+        Args:
+            *args: Positional arguments passed to the parent form.
+            **kwargs: Keyword arguments passed to the parent form. Can include:
+                verified_data (dict, optional): Dictionary containing pre-verified
+                    user data with keys 'deck_name', 'first_name', 'last_name', 'email'.
+        """
+        # Pop verified data from kwargs (if any)
+        self.verified_data = kwargs.pop('verified_data', None)
+        super().__init__(*args, **kwargs)
+        if self.verified_data:
+            # Pre-fill fields from verified session
+            self.fields["first_name"].initial = self.verified_data.get("first_name", "")
+            self.fields["last_name"].initial = self.verified_data.get("last_name", "")
+            self.fields["email"].initial = self.verified_data.get("email", "")
+
+            # Email is already verified so make it immutable
+            self.fields["email"].disabled = True
+
+    def clean_email(self):
+        """
+        Ensure the email field has a valid value during form cleaning.
+
+        Returns the initial verified email if the field is disabled,
+        otherwise returns the cleaned data from user input.
+
+        Returns:
+            str: The email address for the tenant owner.
+        """
+        if self.fields["email"].disabled:
+            email = (self.verified_data or {}).get("email") or self.initial.get("email")
+            if not email:
+                raise forms.ValidationError("Verified e-mail address is required.")
+            return email
+        return self.cleaned_data.get("email")
+
+    def save(self, commit=True):
+        """
+        Save the Tenant and ensure the deck owner User has the correct email and names.
```

Comment:
Why the commit parameter, in what workflow would we set it to false (and do we do that somewhere?)